usaspending_api/common/helpers/fiscal_year_helpers.py

```diff
@@ -193,6 +194,38 @@ def create_full_time_periods(min_date, max_date, group, columns):
     return results
 
 
+def _clean_subaward_spending_over_time_results(subaward_results: List[dict], date_range_type: str) -> List[dict]:
+    """Clean up the spending_over_time Subaward results
+
+    Args:
+        subaward_results: Subaward spending_over_time result for a given time period
+        date_range_type: how the results are split
+            - 'fy', 'quarter', or 'month'
+
+    Returns:
+        List of cleaned Subaward results
+    """
+
+    for result in subaward_results:
+        result["time_period"]["fiscal_year"] = result["time_period"]["fy"]
+        result["aggregated_amount"] = result.get("sub-contract", 0) + result.get("sub-grant", 0)
+
+        result["Contract_Obligations"] = result.get("sub-contract", 0)
+        result["Grant_Obligations"] = result.get("sub-grant", 0)
+
+        result["total_outlays"] = None
+        result["Contract_Outlays"] = None
+        result["Grant_Outlays"] = None
+
+        del result["time_period"]["fy"]
```

Comment:
I was using `pop()` as a way to remove the `sub-contract` and/or `sub-grant` keys that may not exist, without throwing an exception.

This definitely isn't the most explicit way to go about this. I was trying to avoid scanning each dictionary twice (once to see if the key exists and a second time to try and delete the key), but that minor performance saving is probably offset by making the code harder to understand.

Reply:
Updated the code with an `if` check to only delete these keys if they are present. This is easier to understand than the `.pop()` code.